tests/Doctrine/Tests/Common/Persistence/Mapping/ClassMetadataFactoryTest.php

```diff
@@ -159,6 +159,35 @@ public function testWillIgnoreCacheEntriesThatAreNotMetadataInstances()
 
         self::assertSame($metadata, $this->cmf->getMetadataFor('Foo'));
     }
+
+    public function testFallbackMetadataShouldBeCached()
+    {
+        /** @var Cache|PHPUnit_Framework_MockObject_MockObject $cacheDriver */
```

Comment:
We should rather rely on `PHPUnit\Framework\MockObject\MockObject` since it's part of PHPUnit's forward compatibility layer (even though the rest of the test uses the old interface).